src/main/scala/org/broadinstitute/hail/expr/AST.scala

```diff
@@ -213,6 +213,37 @@ object AST extends Positional {
         null
     }
   }
+
+  def evalComposeNumericArray[T](ec: EvalContext, subexpr: AST)(g: (IndexedSeq[T]) => Any)
```

Comment:
This is a good change, let's get it in.  I also think the expr code has gotten completely out of hand.  I was debating if it is even worth adding new evalComposes.  Let's have a meeting with @danking this week and start to flesh out a plan towards HailQL + codegen.